plugins/save-ansi.py

```
@@ -0,0 +1,165 @@
+        for i, char in enumerate(text):
+            grid[y][x + i] = (char, attributes, fg, bg)
+
+    # Generate output
+    output = ''
+    for y in range(max_y + 1):
+        line = ''
+        prev_codes = None
+        for x in range(max_x + 1):
+            cell = grid[y][x]
+            if cell:
+                char, attributes, fg, bg = cell
+                codes = get_escape_codes(attributes, fg, bg)
+                if codes != prev_codes:
+                    line += codes
+                    prev_codes = codes
+                line += char
+            else:
+                if prev_codes != '\033[0m':
+                    line += '\033[0m'
+                    prev_codes = '\033[0m'
+                line += ' '
+        line += '\033[0m'
+        output += line + '\n'
+    return output
+
+def save_ansi(self):
+    """Save the current sheet as an ANSI text file, exporting only rows with empty frame and type."""
+    required_columns = ['x', 'y', 'text', 'color']
+    columns_dict = {col.name: col for col in self.columns}
+    for col_name in required_columns:
+        if col_name not in columns_dict:
+            vd.fail(f'Missing column: {col_name}')
+
+    x_col = columns_dict['x']
+    y_col = columns_dict['y']
+    text_col = columns_dict['text']
+    color_col = columns_dict['color']
+    frame_col = columns_dict.get('frame', None)
+    type_col = columns_dict.get('type', None)
+
+    # Filter rows where both frame and type are empty
+    filtered_rows = []
+    for row in self.rows:
+        frame_value = frame_col.getValue(row) if frame_col else None
+        type_value = type_col.getValue(row) if type_col else None
+        # Consider a value empty if it is None or an empty string
+        if (frame_value is None or frame_value == '') and (type_value is None or type_value == ''):
+            filtered_rows.append(row)
+
+    if not filtered_rows:
+        vd.warning('Drawing is animation; cannot export as ANSI.')
```

In general, use `fail` instead of `warning` + `return`. It's a minor difference but it causes the command to abort (via exception) instead of complete, which has some other consequences (like not being saved to the cmdlog, for instance).